runtime_builders/publish_runtime.py

```diff
@@ -0,0 +1,185 @@
+    'python',
+    'ruby',
+    'nodejs',
+    'golang',
+    'dotnet',
+    'php',
+    'test'
+]
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--infile', '-i', help='file(s) to read')
+    parser.add_argument('--bucket', '-b',
+                        help='GCS bucket to publish runtime to',
+                        default='runtime-builders')
+    parser.add_argument('--language', '-l',
+                        help='the language associated with this builder')
+    args = parser.parse_args()
+
+    if args.infile is None:
+        logging.error('Please specify templated input YAML file.')
+        return 1
+    if args.language is None:
+        logging.error('Please specify language to '
+                      'associate this builder with.')
+        return 1
+    if args.language not in LANGUAGES:
+        logging.error('Invalid language \'{0}\' specified! '
+                      'Options:'.format(args.language))
+        logging.error(LANGUAGES)
+        return 1
+
+    templated_file = _resolve_tags(args.infile)
+    _publish_to_gcs(templated_file, args.language, args.bucket)
+
+
+'''
+Given a templated YAML cloudbuild config file, parse it, resolve image tags
+on each build step's image to the corresponding digest, and write new config
+with fully qualified images to temporary file for upload to GCS.
+
+Keyword arguments:
+config_file -- string representing path to templated cloudbuild YAML config file
+
+Return value:
+path to temporary file containing fully qualified config file, to be
+published to GCS.
+'''
+def _resolve_tags(config_file):
+    with open(config_file, 'r') as infile:
+
+        # try:
+        #   config = f.readlines()
+        #   with open('/tmp/test.txt', 'w') as outfile:
+        #     try:
+        #       for line in config:
+        #         # print line
+        #         # match = re.search(TAG_REGEX, line)
+        #         match = TAG_PATTERN.search(line)
+        #         if match is not None:
+        #           print 'match: {0}'.format(
+        #               match.string[match.start():match.end()])
+        #           # print match.span()
+        #         # if TAG_PATTERN.matches(line):
+        #           outfile.write(_resolve_tag_in_line(line))
+        #         else:
+        #           outfile.write(line)
+        #     finally:
+        #       outfile.close()
+        # finally:
+        #   f.close()
+
+        try:
+            config = yaml.round_trip_load(infile)
+
+            for step in config.get('steps'):
+                image = step.get('name')
+                templated_step = _resolve_tag(image)
+                step['name'] = templated_step
+
+            s = yaml.round_trip_dump(config)
+
+            fd, ofile = tempfile.mkstemp()
+            with open(ofile, 'w') as outfile:
+                outfile.write(s)
+            outfile.close()
+            os.close(fd)
+            return ofile
+
+        except yaml.YAMLError as e:
```

Comment:
ah yeah good call, I just wanted to wrap the error in something a little prettier, but I still want to `sys.exit` here